src/main/java/com/bobvaraioa/kubejspowah/KubeJSPowahPlugin.java

```diff
@@ -1,70 +1,78 @@
+import dev.latvian.mods.kubejs.event.KubeEvent;
+import dev.latvian.mods.kubejs.plugin.KubeJSPlugin;
 import dev.latvian.mods.kubejs.recipe.RecipeKey;
-import dev.latvian.mods.kubejs.recipe.component.ItemComponents;
+import dev.latvian.mods.kubejs.recipe.component.ComponentRole;
+import dev.latvian.mods.kubejs.recipe.component.IngredientComponent;
+import dev.latvian.mods.kubejs.recipe.component.ItemStackComponent;
 import dev.latvian.mods.kubejs.recipe.component.NumberComponent;
 import dev.latvian.mods.kubejs.recipe.schema.RecipeSchema;
-import dev.latvian.mods.kubejs.recipe.schema.RegisterRecipeSchemasEvent;
+import dev.latvian.mods.kubejs.recipe.schema.RecipeSchemaRegistry;
 import net.minecraft.resources.ResourceLocation;
-import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
-import org.checkerframework.checker.units.qual.C;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.material.Fluid;
 import owmii.powah.api.PowahAPI;
 
-public class KubeJSPowahPlugin extends KubeJSPlugin {
+import java.util.List;
+
+public class KubeJSPowahPlugin implements KubeJSPlugin {
     public static EventGroup GROUP = EventGroup.of("PowahEvents");
     public static EventHandler COOLANTS = GROUP.server("registerCoolants", () -> CoolantsEvent.class);
     public static EventHandler HEAT_SOURCE = GROUP.server("registerHeatSource", () -> HeatSourceEvent.class);
     public static EventHandler MAGMATIC_FLUID = GROUP.server("registerMagmaticFluid", () -> MagmaticFluidEvent.class);
 
     interface EnergizingOrbRecipe {
-        RecipeKey<InputItem[]> INPUTS = ItemComponents.UNWRAPPED_INPUT_ARRAY.key("ingredients");
-        RecipeKey<OutputItem> OUTPUT = ItemComponents.OUTPUT.key("result");
-        RecipeKey<Long> ENERGY = NumberComponent.LONG.key("energy");
+        RecipeKey<List<Ingredient>> INPUTS = IngredientComponent.UNWRAPPED_INGREDIENT_LIST.key("ingredients", ComponentRole.INPUT);
+        RecipeKey<ItemStack> OUTPUT = ItemStackComponent.ITEM_STACK.key("result", ComponentRole.OUTPUT);
+        RecipeKey<Long> ENERGY = NumberComponent.LONG.key("energy", ComponentRole.OTHER);
 
         RecipeSchema SCHEMA = new RecipeSchema(INPUTS, OUTPUT, ENERGY);
     }
 
     @Override
-    public void registerRecipeSchemas(RegisterRecipeSchemasEvent event) {
-        event.register(new ResourceLocation("powah:energizing"), EnergizingOrbRecipe.SCHEMA);
+    public void registerRecipeSchemas(RecipeSchemaRegistry event) {
+        event.register(ResourceLocation.fromNamespaceAndPath("powah", "energizing"), EnergizingOrbRecipe.SCHEMA);
     }
 
     @Override
-    public void registerEvents() {
-        GROUP.register();
+    public void registerEvents(EventGroupRegistry registry) {
+        registry.register(GROUP);
     }
 
-    static class CoolantsEvent extends EventJS {
+    static class CoolantsEvent implements KubeEvent {
```

Comment:
These are not the correct methods that the API uses? 
See https://github.com/Technici4n/Powah/pull/161/commits/a7acf989ca46df48b72f46353173db71eb5ee1a7 for how this changed. It should use data maps instead